### PR TITLE
[ADD] 16.0 company_dependent_flag module

### PR DESCRIPTION
**Pull request by bealdav**
cpny_style

### CODE REVIEW

File: company_dependent_flag/models/base.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            self._update_company_dependent_css(arch)
+        return arch, view
+
+    def _update_company_dependent_css(self, arch):
+        cpny_dep_fields = [
+            x for x, y in self.env[self._name]._fields.items() if y.company_dependent
+        ]
+        for field_name in cpny_dep_fields:
+            for field in arch.xpath(f"//field[@name='{field_name}']"):
+                field.attrib["class"] = self._get_company_dependent_css_class()
```
Comment (bealdav) [inline company_dependent_flag/models/base.py]: TODO need to add class, not to replace

**Review by PaulGoubert — APPROVED** [review]
Work fine.   Thx @bealdav

**Comment by bealdav** [comment]
please @gurneyalex @JordiBForgeFlow could you have a look here please ?

**Review by kevinkhao — APPROVED** [review]
maybe aesthetic aspect could be improved in css but LGTM

File: company_dependent_flag/models/base.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            field_name
+            for field_name, field_rec in self.env[self._name]._fields.items()
+            if field_rec.company_dependent
+        ]
+        for field_name in cpny_dep_fields:
+            for field in arch.xpath(f"//field[@name='{field_name}']"):
+                fa = self._get_company_dependent_css_class()
+                field.attrib[
+                    "class"
+                ] = f"{fa} {field.attrib.get('class') or ''}".strip()
```
Comment (legalsylvain) [inline company_dependent_flag/models/base.py:28]:
```suggestion
                class = self._get_company_dependent_css_class()
                field.attrib[
                    "class"
                ] = f"{class} {field.attrib.get('class', '')}".strip()
```

Comment (legalsylvain) [inline company_dependent_flag/models/base.py]:
```suggestion
                classes = field.attrib.get("class", "").split(" ")
                classes += self._get_company_dependent_css_class()
                field.attrib["class"] = " ".join(set(classes))

File: company_dependent_flag/models/base.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    def _get_company_dependent_css_class(self):
+        """Inherit to apply your own class"""
+        return "fa fa-building-o"
```
Comment (legalsylvain) [inline company_dependent_flag/models/base.py]:
```suggestion
        return ["fa", "fa-building-o"]
```

**Review by legalsylvain — COMMENTED** [review]
minor suggestion. avoid to have duplicated classes. ("fa" for instance)

**Comment by bealdav** [comment]
Thanks a lot @legalsylvain

**Pull request closed without merging**